### PR TITLE
CSI guideline

### LINKED ISSUE #2333: [FEATURE] Commenting Showing Intent (CSI)
This is an attempt to increase the quality of our internal documentation (or, in many cases, ensuring that there is one).

To understand what this is, visit the website of this standard: https://standards.mousepawmedia.com/en/stable/csi.html

In order to close this issue, we need to

- [ ] add CSI to contrib guidelines
- [ ] add CSI to PR template
- [ ] do some initial CSI comments in the project

It will be probably too much work to full on comment the whole project in one go, but whenever we touch a file, we should make sure to comment that in.

### PR DESCRIPTION
**Pull request by Poolitzer**
### Checklist for PRs

- [ ] Added `.. versionadded:: version`, `.. versionchanged:: version` or `.. deprecated:: version` to the docstrings for user facing changes (for methods/class descriptions, arguments and attributes)
- [ ] Created new or adapted existing unit tests
- [ ] Added myself alphabetically to `AUTHORS.rst` (optional)

closes #2333

### CODE REVIEW

File: .github/CONTRIBUTING.rst [hunk 1]
```diff
@@ -186,6 +166,49 @@ Here's how to make a one-off code change.
 
 7. **Celebrate.** Congratulations, you have contributed to ``python-telegram-bot``!
 
+Documenting
+===========
+
+The documentation of this project is separated in two sections: User facing and dev facing.
+
+User facing docs are hosted at `RTD`_. They are the main way the users of our library are supposed to get information about the objects. They don't care about the technical side, they just want to know
```
Comment (Bibo-Joshi) [inline .github/CONTRIBUTING.rst:174]:
```suggestion
User facing docs are hosted at `RTD`_. They are the main way the users of our library are supposed to get information about the objects. They don't care about the internals, they just want to know
```

File: .github/CONTRIBUTING.rst [hunk 2]
```diff
@@ -186,6 +166,49 @@ Here's how to make a one-off code change.
...
+what they have to pass to make it work, what it actually does. You can/should provide examples for non obvious cases (like the Filter module), and notes/warnings.
+
+Dev facing, on the other side, is for the devs/maintainers of this project. This is a big part of the documentation we have neglected thus far, but not anymore! These
```
Comment (Bibo-Joshi) [inline .github/CONTRIBUTING.rst:177]:
```suggestion
Dev facing, on the other side, is for the devs/maintainers of this project. These
```

File: .github/CONTRIBUTING.rst [hunk 3]
```diff
@@ -186,6 +166,49 @@ Here's how to make a one-off code change.
...
+
+   $ sphinx-build docs/source docs/build/html
+
+Once the process terminates, you can view the built documentation by opening ``docs/build/html/index.html`` with a browser.
+
+- Add ``.. versionadded:: version``, ``.. versionchanged:: version`` or ``.. deprecated:: version`` to the associated documentation of your changes, depending on what kind of change you made. This only applies if the change you made is visible to an end user. The directives should be added to class/method descriptions if their general behaviour changed and to the description of all arguments & attributes that changed.
+
+Dev facing documentation
+------------------------
+We adhere to the `CSI`_ standard. Again, remember, this documentation is not fully implemented in the project, yet. We still ask you to add it to your code changes and this way, we will slowly document the whole project this way.
```
Comment (Bibo-Joshi) [inline .github/CONTRIBUTING.rst:206]:
```suggestion
We adhere to the `CSI`_ standard. This documentation is not fully implemented in the project, yet, but new code changes should comply with the `CSI` standard.
```

File: .github/CONTRIBUTING.rst [hunk 4]
```diff
@@ -186,6 +166,49 @@ Here's how to make a one-off code change.
...
+The idea behind this is to make it very easy for you/a random maintainer or even a totally foreign person to drop anywhere into the code and more or less immediately understand what a particular line does. This will make it easier
+for new or old devs who are out of the loop to make relevant changes if said lines don't do what they are supposed to. Even though implementing this will mean a lot of comments, we still believe in it being a very meaningful addition to our project.
```
Comment (Bibo-Joshi) [inline .github/CONTRIBUTING.rst:208]:
```suggestion
for new to make relevant changes if said lines don't do what they are supposed to.
```

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -6,6 +6,7 @@ Hey! You're PRing? Cool! Please have a look at the below checklist. It's here to
 
 - [ ] Added `.. versionadded:: version`, `.. versionchanged:: version` or `.. deprecated:: version` to the docstrings for user facing changes (for methods/class descriptions, arguments and attributes)
 - [ ] Created new or adapted existing unit tests
+- [ ] Documented code dev facing
```
Comment (Bibo-Joshi) [inline .github/pull_request_template.md:9]:
```suggestion
- [ ] Documented code changes according to the [CSI standard](https://standards.mousepawmedia.com/en/stable/csi.html)
```

**Review by Bibo-Joshi — COMMENTED** [review]
Like the overall structure 👍 Just added a few comments - IMHO we don't need to justify ourselfs in the contrib guide …

File: .github/pull_request_template.md (hunk 5 above)
Comment (Poolitzer) [reply .github/pull_request_template.md:9]: upps

**Pull request merged**